clients/php/tests/ZawgyiDetectorTest.php

```diff
@@ -0,0 +1,110 @@
+    {
+        $strongZawgyi = "အျပည္ျပည္ဆိုင္ရာ လူ႔အခြင့္အေရး ေၾကညာစာတမ္း";
+        $this->assertGreaterThan(0.999, $this->detector->getZawgyiProbability($strongZawgyi), "Strong Zawgyi");
+    }
+
+    /** @tests */
+    public function ignore_numerals()
+    {
+        // Digits/numerals are ignored and treated like non-Myanmar code points.
+        $this->assertEquals(INF, $this->detector->getZawgyiProbability("၉၆.၀ kHz"));
+        $this->assertEquals($this->detector->getZawgyiProbability("ဒဂုန်"), $this->detector->getZawgyiProbability("၂၄၀၉ ဒဂုန်"));
+    }
+
+    /** @tests */
+    public function difficult_code_points()
+    {
+        /* These are strings that the detector has gotten wrong. They mostly render in both Zawgyi and
+         * Unicode, but the words they spell make sense in one encoding but not in the other. As the
+         * detector improves, change this tests case with scores to match the new output.
+         */
+        $cases = [
+            // STRINGS IDENTICAL IN UNICODE/ZAWGYI
+            // TODO: These should return scores in the middle range.
+            ["အသံကို အစားထိုးလိုပါသလား။", 0.995], // Truth: Identical in U/Z
+            ["နမူနာ", 0.90], // Truth: Identical in U/Z
+            [" ဦး", 0.995], // Truth: Identical in U/Z
+
+            // UNICODE STRINGS WITH HIGH ZAWGYI SCORES
+            // TODO: Fix detection so that these get low Zawgyi scores.
+            ["အစားထိုး အထူးအက္ခရာ", 0.995], // Truth: Unicode (confirmed by yinmay@)
+            ["ယခု မိုးရွာနေပါသလား။", 0.995], // Truth: Unicode (confirmed by yinmay@)
+            ["အခြား", 0.63], // Truth: Unicode (confirmed by yinmay@)
+
+            // DIFFICULT STRINGS THAT DETECT CORRECTLY
+            // Changes to the detector should not significantly change these scores.
+            ["ကာမစာအုပ္မ်ား(ေစာက္ပတ္စာအုပ္မ်ား)", 1.0], // Truth: Zawgyi (confirmed by yinmay@)
+            ["ညႇပ္စရာမလို", 1.0], // Truth: Zawgyi (confirmed by yinmay@)
+        ];
+
+        foreach ($cases as $case) {
+            $input    = $case[0];
+            $expected = $case[1];
+            $actual   = $this->detector->getZawgyiProbability($input);
+            $delta    = 0.005;
+            $this->assertEquals($expected, $actual, $input, $delta);
+        }
+    }
+
+    /** @tests */
+    public function compatibility()
+    {
+        $fileName = __DIR__ . '/../resources/compatibility.tsv';
+        $row = 1;
+        if (($handle = fopen($fileName, "r")) !== FALSE) {
+            // omitting the length parameter by setting zero
+            while (($data = fgetcsv($handle, 0, "\t")) !== FALSE) {
+                $expected = $data[0] === "-Infinity" ? INF : (double) $data[0];
+                $actual = $this->detector->getZawgyiProbability($data[1]);
+                // Since php natural logarithmic calculation (exp()) is a little bit far from exact value
+                // we have to put very very small delta point
+                $delta = 0.00000001;
```

Comment:
After you remove the `round` on line 71 of BinaryMarkov.php, try removing the delta here.  That might be the problem.  Otherwise, it is fine if you need to keep the delta here.